.github/workflows/build_and_push.yml

```diff
@@ -0,0 +1,55 @@
+name: Build and Push Container to ECR
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+env:
+  GITHUB_SHA: ${{ github.sha }}
+  REGISTRY: 545719101351.dkr.ecr.ca-central-1.amazonaws.com/next-auth-sandbox-wm
+
+permissions:
+  id-token: write
+  contents: read
+  pull-requests: write
+  actions: write
+  checks: write
+  statuses: write
+  security-events: write
```

Comment:
If you end up switching to using AWS access keys for authentication, you can remove this block.  It's only needed for OIDC to set this:
```
permissions:
  id-token: write
```